packages/expo-mail-composer/src/openClientAsync.ios.ts

```diff
@@ -0,0 +1,86 @@
+import { canOpenURL, openURL } from 'expo-linking';
```

Comment:
To be honest, I'm a little bit reluctant to adding a dependency on any other module. Although `expo-linking` is not a native module, it also depends on `expo-constants`. Dependencies across RN libs is generally a problem, because it could lead to some unexpected behavior – there might be two versions in `node_modules`, but only one can be linked and the JS code from both might get included in the JS bundle. Is there any way to avoid it? I think we can do it natively. Since we don't need to hardcode these mail clients and their schemes on Android, hardcoding it on iOS instead of JS sounds fine.